statusnet/library/twitteroauth.php

```diff
@@ -60,60 +73,49 @@ function requestTokenURL()
 		return 'https://api.twitter.com/oauth/request_token';
 	}
 
-	/**
-	 * Debug helpers
-	 */
-	function lastStatusCode()
-	{
-		return $this->http_status;
-	}
-
-	function lastAPICall()
-	{
-		return $this->last_api_call;
-	}
-
-	/**
-	 * construct TwitterOAuth object
-	 */
-	function __construct($consumer_key, $consumer_secret, $oauth_token = NULL, $oauth_token_secret = NULL)
+	function __construct($consumer_key, $consumer_secret, $oauth_token = null, $oauth_token_secret = null)
 	{
 		$this->sha1_method = new OAuthSignatureMethod_HMAC_SHA1();
 		$this->consumer = new OAuthConsumer($consumer_key, $consumer_secret);
 		if (!empty($oauth_token) && !empty($oauth_token_secret)) {
-			$this->token = new OAuthConsumer($oauth_token, $oauth_token_secret);
+			$this->token = new OAuthToken($oauth_token, $oauth_token_secret);
```

Comment:
What I don't understand: Why had it worked before?

Reply:
Both classes have identical token/password public members, so it didn't matter which class you instantiate with just a token and password, the data could be retrieved the same way from both objects. With type-hints it isn't possible anymore.